contracts/token/erc721/extensions/ERC721HybridMinting.sol

```diff
@@ -0,0 +1,226 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: MIT
+
+import { ERC721 } from "@openzeppelin/contracts/token/ERC721/ERC721.sol";
+import { ERC721Psi, ERC721PsiBurnable } from "../erc721psi/ERC721PsiBurnable.sol";
+
+/*
+This contract allows for minting with one of two strategies:
+- ERC721: minting with specified tokenIDs (inefficient)
+- ERC721Psi: minting in batches with consecutive tokenIDs (efficient)
+
+All other ERC721 functions are supported, with routing logic depending on the tokenId. 
+*/
+
+abstract contract ERC721HybridMinting is ERC721PsiBurnable, ERC721 {
```

Comment:
can we have this extend `ERC721Royalty` instead of `ERC721` since `ERC721Royalty` already extends `ERC721`?